src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -47,59 +75,152 @@ private CoseHeaderMap(bool isReadOnly)
             }
         }
 
+        /// <summary>
+        /// Gets the CBOR encoded value associated with the specified label, as a signed integer.
+        /// </summary>
+        /// <param name="label">The label of the value to get.</param>
+        /// <returns>The decoded value associated with the specified label, as a signed integer.</returns>
+        /// <exception cref="InvalidOperationException">Decodification failed, see the inner exception for the exact exception that was thrown.
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public int GetValueAsInt32(CoseHeaderLabel label) => _headerParameters[label].GetValueAsInt32();
 
+        /// <summary>
+        /// Gets the CBOR encoded value associated with the specified label, as a text string.
+        /// </summary>
+        /// <param name="label">The label of the value to get.</param>
+        /// <returns>The decoded value associated with the specified label, as a text string.</returns>
+        /// <exception cref="InvalidOperationException">Decodification failed, see the inner exception for the exact exception that was thrown.
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public string GetValueAsString(CoseHeaderLabel label) => _headerParameters[label].GetValueAsString();
 
+        /// <summary>
+        /// Gets the CBOR encoded value associated with the specified label, as a byte string.
+        /// </summary>
+        /// <param name="label">The label of the value to get.</param>
+        /// <returns>The decoded value associated with the specified label, as a byte string.</returns>
+        /// <exception cref="InvalidOperationException">Decodification failed, see the inner exception for the exact exception that was thrown.
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public byte[] GetValueAsBytes(CoseHeaderLabel label) => _headerParameters[label].GetValueAsBytes();
 
+        /// <summary>
+        /// Gets the CBOR encoded value associated with the specified label, as a byte string.
+        /// </summary>
+        /// <param name="label">The label of the value to get.</param>
+        /// <param name="destination">The buffer in which to write the decoded value.</param>
```

Comment:
Small suggestion. The `<summary>` says "Gets the CBOR _encoded_ value" but the param documentation says "to write the _decoded_ value.". 

It would be nice if there was agreement in the language for this API about whether the value is encoded or decoded.

Reply:
The value is decoded. What I mean in the summary is that we will attempt to _decode_ the "CBOR-encoded value" as a "CBOR byte string".
Maybe we can just say "Gets the _value_ associated with the specified label, as a byte string"

Reply:
The ASN.1 docs all just tiptoe around this with a very vague

```suggestion
        /// <summary>
        /// Gets the CBOR encoded value associated with the specified label, as a byte string.
        /// </summary>
        /// <param name="label">The label of the value to get.</param>
        /// <param name="destination">The buffer in which to write.</param>
```

At least, that's what AsnWriter.Encode(Span) and AsnDecoder.TryReadCharacterStringBytes both did.